### ISSUE TITLE
 TypeError: create_repo() got an unexpected keyword argument 'organization'

### ISSUE THREAD

**Issue by SzaboGergo01** [huggingface/transformers#22621]
### System Info

My environment:
```
- `transformers` version: 4.27.4
- Platform: Linux-5.10.147+-x86_64-with-glibc2.31
- Python version: 3.9.16
- Huggingface_hub version: 0.13.3
- PyTorch version (GPU?): 2.0.0+cu118 (True)
- Tensorflow version (GPU?): 2.12.0 (True)
- Flax version (CPU?/GPU?/TPU?): 0.6.8 (gpu)
- Jax version: 0.4.7
- JaxLib version: 0.4.7
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>
```

### Information

- [X] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

```python
model.save_to_hub("sent_bert", 
                  organization="gszabo",
                  train_datasets=["gszabo/sentence-compression"],
                  exist_ok=True, 
                  )
```
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-13-cbde1c79e18f>](https://localhost:8080/#) in <cell line: 1>()
----> 1 model.save_to_hub("sent_bert", 
      2                   organization="gszabo",
      3                   train_datasets=["gszabo/sentence-compression"],
      4                   exist_ok=True,
      5                   )

1 frames
[/usr/local/lib/python3.9/dist-packages/sentence_transformers/SentenceTransformer.py](https://localhost:8080/#) in save_to_hub(self, repo_name, organization, private, commit_message, local_model_path, exist_ok, replace_model_card, train_datasets)
    465 
    466         endpoint = "https://huggingface.co/"
--> 467         repo_url = HfApi(endpoint=endpoint).create_repo(
    468                 token,
    469                 repo_name,

[/usr/local/lib/python3.9/dist-packages/huggingface_hub/utils/_validators.py](https://localhost:8080/#) in _inner_fn(*args, **kwargs)
    118             kwargs = smoothly_deprecate_use_auth_token(fn_name=fn.__name__, has_token=has_token, kwargs=kwargs)
    119 
--> 120         return fn(*args, **kwargs)
    121 
    122     return _inner_fn  # type: ignore

TypeError: create_repo() got an unexpected keyword argument 'organization'
```

### Expected behavior

I used Google Colab, and I wanted to fine-tuned a sentence-bert (`sentence_transformers`) with my own data and then I wanted to push to the HF-hub. After that, I created a public repo on the hub in my account under the name `sent_bert`, before that I also logged in to Colab with `notebook_login()`. Also I created the model and then I used the `save_to_hub()` function when got the error.

I tried to use the `push_to_hub()` function as well, but that doesn't support the sentence transformer models. Has anyone encountered something similar or do you know a solution to this?

Anyway, I pretty much followed [these steps](https://huggingface.co/blog/how-to-train-sentence-transformers):

**Comment by sgugger** [comment]
You need to update your installation of `huggingface_hub`: `pip install --upgrade huggingface_hub`.

**Comment by SzaboGergo01** [comment]
I tried to update but got the same error message.
Anyway, I'm checking now to see if it's a new update, I mean it came out yesterday, but no, it doesn't work either way.

**Comment by sgugger** [comment]
Oh I'm sorry, that's a deprecated argument. You need to pass the organization with the model name now:
`model.save_to_hub("gszabo/sent_bert", ...)`

**Comment by SzaboGergo01** [comment]
Unfortunately, it's not good either...

```python
import huggingface_hub
huggingface_hub.__version__
```
And its version: 0.13.4

**Comment by sgugger** [comment]
What is the error message you are getting?

**Comment by Thabet-Chaaouri** [comment]
I am getting the same error for the same environment

**Comment by sgugger** [comment]
Yes, you need to remove the `organization` argument, it is not accepted anymore. The organization should be part of your repo ID now.

**Comment by sgugger** [comment]
Also please open issues related to sentence-transformers in that repo, there is only so much I can do to help since I don't know it at all and don't maintain it. Closing it here, if you have further issues please go [there](https://github.com/UKPLab/sentence-transformers) :-)

**Comment by sagnik** [comment]
I think the problem here is a version mismatch between `transformers` and `huggigface_hub`. 

For eg, a `transformers` version 4.17.0, will not work with a `huggingface_hub` version of 0.14.1. 

Specifically, [L2954](https://github.com/huggingface/transformers/blob/v4.17.0/src/transformers/file_utils.py#L2954) has the offending call. 

So we have to check when this tag was released, for 4.17.0, the date is March 3, 2022. The appropriate `huggingface_hub` version is probably `0.5.0`, but then you would have to downgrade `datasets`, which depending on the version, might have a wrong filter implementation ([this issue](https://github.com/huggingface/datasets/pull/2947)).

In summary, if you are not updated to the latest transformers + datasets + huggingface_hub, and using all of them, you might break your code.
